### PR TITLE
docs: improve prerequisites for OSX users

### PR DESCRIPTION
**Pull request by jeluard**
**Motivation**

Improves DX for OSX users

**Description**

Add some extra prerequisites for OSX users so that they can build/test `lodestar`.
`python` is required for mac silicon users as https://github.com/ChainSafe/blst-ts doesn't provide a prebuild for those CPUs

### CODE REVIEW

**Review by wemeetagain — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -28,8 +28,7 @@
 
 ## Prerequisites
 
-- :gear: [NodeJS](https://nodejs.org/) (LTS)
-- :toolbox: [Yarn](https://yarnpkg.com/)
+First follow the prerequisites list [here](./CONTRIBUTING.md#prerequisites)
```
Comment (nflaig) [inline README.md]: I am not a fan of referencing the contribution.md here. We have references to that already in the readme and contributors should look at that anyways. The readme is more of a quickstart to get Lodestar to run and for most operating systems (Linux & WSL) it will just work.

Let's get some more opinions on this

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -7,6 +7,13 @@ Thanks for your interest in contributing to Lodestar. It's people like you that
 - :gear: [NodeJS](https://nodejs.org/) (LTS)
 - :toolbox: [Yarn](https://yarnpkg.com/)
 
+### OSX Specifics
+
+When using OSX a couple extra prerequisites are required.
+
+* python
+* coreutils (e.g. via `brew install coreutils`)
```
Comment (nflaig) [inline CONTRIBUTING.md]:
```suggestion
- python
- coreutils (e.g. via `brew install coreutils`)
```

File: packages/db/test/unit/controller/level.test.ts [hunk 3]
```diff
@@ -144,7 +144,7 @@ describe("LevelDB controller", () => {
           return "gdu";
         }
       } catch {
-        /* no-op */
+        console.error('Cannot find correct gdu command');
```
Comment (nflaig) [inline packages/db/test/unit/controller/level.test.ts]:
```suggestion
        console.error("Cannot find correct gdu command");
```

**Review by nflaig — COMMENTED** [review]
Not sure about adding os specifics like this, @matthewkeil probably has some thoughts. Does the docs rework maybe address this is a nicer way?

File: README.md (hunk 1 above)
Comment (jeluard) [reply README.md]: Makes sense, I can revert this. The idea was to not duplicate now your point makes sense.
I would expect the build to fail for mac users with mac silicon due to missing `python` though.

Comment (matthewkeil) [reply README.md]: I tend to agree that this should not be referenced here.  Those two docs are for different purposes and have different audiences

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -7,6 +7,13 @@ Thanks for your interest in contributing to Lodestar. It's people like you that
 - :gear: [NodeJS](https://nodejs.org/) (LTS)
 - :toolbox: [Yarn](https://yarnpkg.com/)
 
+### MacOS Specifics
+
+When using MacOS a couple extra prerequisites are required.
```
Comment (matthewkeil) [inline CONTRIBUTING.md]:
```suggestion
When using MacOS, there are a couple of extra prerequisites that are required.
```

File: packages/db/test/unit/controller/level.test.ts [hunk 5]
```diff
@@ -144,7 +144,8 @@ describe("LevelDB controller", () => {
           return "gdu";
         }
       } catch {
-        /* no-op */
+        /* eslint-disable no-console */
+        console.error("Cannot find correct gdu command");
```
Comment (matthewkeil) [inline packages/db/test/unit/controller/level.test.ts]: How do you feel about saying

```suggestion
        console.error("Cannot find gdu command, falling back to du");
```

**Review by matthewkeil — CHANGES_REQUESTED** [review]
Just a couple of nits but overall strong start!  Welcome to the team 🚀

File: README.md (hunk 1 above)
Comment (nflaig) [inline README.md]: Looks good to me, let's revert this change and we can merge it

Comment (jeluard) [reply README.md]: Done

**Review by nflaig — APPROVED** [review]

**Review by matthewkeil — APPROVED** [review]
LGTM!  🎸

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -7,6 +7,13 @@ Thanks for your interest in contributing to Lodestar. It's people like you that
 - :gear: [NodeJS](https://nodejs.org/) (LTS)
 - :toolbox: [Yarn](https://yarnpkg.com/)
 
+### MacOS Specifics
+
+When using MacOS, there are a couple of extra prerequisites that are required.
+
+- python
+- coreutils (e.g. via `brew install coreutils`)
```
Comment (nazarhussain) [inline CONTRIBUTING.md:15]: I used MacOS regularly for years but never intentionally installed these to start with lodestar (those packages may be installed with git or other tool). 

Out of curiosity when did you encounter error for these in reference to lodestar?

Comment (nflaig) [reply CONTRIBUTING.md:15]: Maybe it's related to the node version? This might not happen if there are pre-builts available for blst

Comment (jeluard) [reply CONTRIBUTING.md:15]: This is required for unit tests. See this [check](https://github.com/ChainSafe/lodestar/blob/unstable/packages/db/test/unit/controller/level.test.ts#L140)

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.13.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.13.0) :tada:
